text-generation-inference/server/text_generation_server/cli.py

```diff
@@ -62,8 +62,28 @@ def serve(
 
     # Downgrade enum into str for easier management later on
     quantize = None if quantize is None else quantize.value
-    dtype = None if dtype is None else dtype.value
-    server.serve(model_id, revision, dtype, uds_path)
+    dtype = "bfloat16" if dtype is None else dtype.value
+
+    logger.info("CLI SHARED = {} DTYPE = {}".format(sharded, dtype))
+
+    if sharded:
+        num_shard = int(os.getenv("WORLD_SIZE", "1"))
+        logger.info("CLI SHARED = {}".format(num_shard))
+        import subprocess
+
+        cmd = f"deepspeed --num_nodes 1 --num_gpus {num_shard} --no_local_rank /usr/src/server/text_generation_server/tgi_service.py --model_id {model_id} --revision {revision} --sharded {sharded} --dtype {dtype} --uds_path {uds_path}"
```

Comment:
Using hard-coded path is not the best solution. What if the path will be different?

Reply:
+1, this should be changed